src/docbuilder/rustwide_builder.rs

```diff
@@ -304,22 +293,25 @@ impl RustwideBuilder {
                     self.copy_docs(
                         &build.host_target_dir(),
                         local_storage.path(),
-                        if in_target { &res.target } else { "" },
+                        "",
                         true,
                     )?;
 
-                    if in_target {
-                        // Then build the documentation for all the targets
-                        for target in TARGETS {
-                            debug!("building package {} {} for {}", name, version, target);
-                            self.build_target(
-                                target,
-                                &build,
-                                &limits,
-                                &local_storage.path(),
-                                &mut successful_targets,
-                            )?;
+                    successful_targets.push(res.target.clone());
+                    // Then build the documentation for all the targets
+                    for target in &metadata.extra_targets {
```

Comment:
To merge this without breaking the crates, we could gate the new behavior on the `DOCS_RS_ONLY_BUILD_DEFAULT_TARGET` environment variable: if it's not set we iterate over `TARGETS` instead. This would avoid this PR rotting over time if we wait until February/March.

Reply:
The actual breaking change should be very small, just changing `for target in TARGETS {` to `for target in &metadata.extra_targets {`. This will be easier to see when I break up the PR. I think an environment variable would be helpful for local development + testing though so I'll make the change.

Reply:
Added in https://github.com/rust-lang/docs.rs/pull/532/commits/2b8a99a768dd8d65767152eefb5c0676436469fc